lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -925,6 +925,22 @@ defmodule Kernel.ErrorsTest do
       'case [] do; [] when Hello.something_that_does_not_exist == [] -> :ok; end'
   end
 
+  test "invalid :erlang call on match" do
+    assert_compile_fail CompileError,
+      "nofile:1: cannot invoke remote function :erlang.make_ref/0 inside match",
+      'case [] do; :erlang.make_ref() -> :ok; end'
+
+    assert_compile_fail CompileError,
+      "nofile:1: cannot invoke remote function :erlang.self/0 inside match",
+      'case [] do; self() -> :ok; end'
+  end
+
+  test "invalid :erlang call on guard" do
+    assert_compile_fail CompileError,
+      "nofile:1: cannot invoke remote function :erlang.make_ref/0 inside match",
+      'case [] do; :erlang.make_ref() -> :ok; end'
```

Comment:
I know the rest of the file is like this so that might be the reason, but why are we using `;`s here instead of:

```elixir
case [] do :erlang.make_ref() -> :ok end
```

Is there a specific reason or just No Particular Reason™?